src/engine/trade.ts

```diff
@@ -2,27 +2,65 @@ import { ExchangeProvider } from '@exchange'
 import { TradeStore, MarkerStore } from '@store'
 import { sleep, logger } from '@util'
 
+enum SYNC_STATE {
+  STOPPED,
+  SYNCING,
+  READY,
+}
+
 export class TradeEngine {
+  private readonly symbols: Map<string, SYNC_STATE> = new Map()
   private readonly scanType: 'back' | 'forward'
 
   private readonly tradeStore = TradeStore.instance
   private readonly markerStore = MarkerStore.instance
 
-  constructor(private readonly exchange: string, private readonly exchangeProvider: ExchangeProvider, private readonly tradePollInterval: number) {
+  constructor(private readonly exchangeProvider: ExchangeProvider, private readonly exchange: string, private readonly tradePollInterval: number) {
     this.scanType = this.exchangeProvider.getScan(this.exchange)
   }
 
-  public async scan(symbol: string, days: number) {
+  public isReady(symbol: string) {
+    return this.symbols.get(symbol) === SYNC_STATE.READY
+  }
+
+  public async start(symbol: string, days: number) {
+    if (this.symbols.has(symbol) && this.symbols.get(symbol) !== SYNC_STATE.STOPPED) return
+
+    logger.info(`Trade sync for ${this.exchange} on ${symbol} started.`)
+    this.symbols.set(symbol, SYNC_STATE.SYNCING)
+    this.tradeStore.addSymbol({ exchange: this.exchange, symbol })
+
+    await this.scan(symbol, days)
+    this.symbols.set(symbol, SYNC_STATE.READY)
```

Comment:
Before setting this you have to check the state. Otherwise your break-condition in `this.scan` is not valid.
If you stop the TradeEngine while its scanning ill return, and will set itself to Ready afterwards. I think thats not valid.